src/canvas-extractor/edfi_canvas_extractor/client_graphql.py

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+import sqlalchemy
+
+from pandas import DataFrame
+from typing import List, Tuple
+
+from canvasapi.course import Course
+from canvasapi.section import Section
+
+from edfi_canvas_extractor.graphql.extract import Extract
+from edfi_canvas_extractor.graphql import (
+    courses as coursesGQL,
+    sections as sectionsGQL,
+)
+from edfi_canvas_extractor.mapping import (
+    sections as sectionsMap,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def extract_courses(
+    gql: Extract,
```

Comment:
`Extract` is one of those strange English words that is both a verb and a noun. It took me a little while to understand what I was seeing, because initially I assumed that `Extract` is the _result_, that is, the X in "I extracted X". As a noun this didn't make sense to me. Once I thought of it as a verb, it made more sense! Suggestion: rename `Extract` to `GraphQLExtractor` or something like that. "Extractor" makes more clear that this is the module that "extracts from GraphQL".

Reply:
Got the same issue naming variables, etc. even on spanish so I am ok with that name of class, Working on ...